tests/test_herbstclient.py

```diff
@@ -139,3 +139,83 @@ def test_lastarg_only(hlwm, repeat):
     assert proc.stdout.read().splitlines() == expected_lines
     proc.wait(20)
     assert proc.returncode == 0
+
+
+def test_version_without_hlwm_running():
+    for cmd in [[HC_PATH, '-v'], [HC_PATH, '--version']]:
+        proc = subprocess.run(cmd,
+                              stdout=subprocess.PIPE,
+                              universal_newlines=True)
+        # test that the output ends with a newline
+        assert re.match('^herbstclient.*\n$', proc.stdout)
+
+
+def test_quiet_without_hlwm_running():
+    for flag in ['-q', '--quiet']:
+        cmd = [HC_PATH, flag, '-q', 'echo', 'test']
+        proc = subprocess.run(cmd,
+                              stdout=subprocess.PIPE,
+                              stderr=subprocess.PIPE,
+                              universal_newlines=True)
+        assert proc.returncode != 0
+        assert proc.stdout == ''
+        assert proc.stderr == ''
+
+
+def test_quiet_with_hlwm_running(hlwm):
+    # even with --quiet, herbstclient must produce output
+    for flag in ['-q', '--quiet']:
+        cmd = [HC_PATH, flag, '-q', 'echo', 'test']
```

Comment:
Why is -q in the parameters, shouldn't flag cover this already?

Reply:
Same for the test above

Reply:
I've looked at the coverage report and `-q` was not handled. I think the issue was the wrong parameter (it's fixed by #1206). Also, this case ensures that the stderr is really empty (i.e. that there is no other error).

Reply:
Yes, but I thought the for-loop is there for iterating over -q and --quiet?
Now cmd is called with ['-q', '-q'] and then ['--quiet', 'q']. 

Maybe I'm overlooking or misunderstanding something

Reply:
Aah, you're completely right. I understood your question entirely wrong (and thus started looking for other test cases where `--quiet` is mentioned...). I've fixed it, sorry for the confusion :)

Reply:
Thanks! But a nice side-effect of uncovering a different testing error :D